### DISCUSSION TITLE
Containerized Rustdesk Pro does not allow login

### DISCUSSION THREAD

**Discussion by MickMaster** [rustdesk/rustdesk-server-pro#347] [Q&A]
Hi,

- I have Rustdesk-Pro installed in a Docker Container
- It ran fine with HTTP
- Now I am using NGINX to access Port 21114 via reverse proxy
- The client acces from the installed client is still working as it should 
- If I want to login into the web-interface I get a "login failed"
- The browser console says a 404 for accessing https://domain.com/api/login-options and /api/login. 

Here is my NGIX config:

```
location /rustdesk/ {
	proxy_pass http://127.0.0.1:21114/;
	proxy_set_header Host $host;
	proxy_set_header X-Real-IP $remote_addr;
	proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
	proxy_set_header X-Forwarded-Proto $scheme;

	# Remove the trailing slash to avoid double slashes
	rewrite ^/rustdesk/(.*) /$1 break;

	# Handle redirects
	proxy_redirect / /rustdesk/;

	# Debug headers
	add_header X-Debug-Message "Proxying to RustDesk" always;

	# Use the custom log format
	access_log /var/log/nginx/rustdesk_access.log;
	error_log /var/log/nginx/rustdesk_error.log;

	add_header 'Access-Control-Allow-Origin' '*';
	add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS';
	add_header 'Access-Control-Allow-Headers' 'DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range';
}

location /rustdesk/static/ {
	proxy_pass http://127.0.0.1:21114/static/;
	proxy_set_header Host $host;
	proxy_set_header X-Real-IP $remote_addr;
	proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
	proxy_set_header X-Forwarded-Proto $scheme;
}

location /rustdesk/api {
	proxy_pass http://127.0.0.1:21114/api;
	proxy_set_header Host $host;
	proxy_set_header X-Real-IP $remote_addr;
	proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
	proxy_set_header X-Forwarded-Proto $scheme;
}

# Handle root redirect
location = /rustdesk {
	return 301 $scheme://$host$request_uri/;
}

````
Here are my questions:

* What is wrong with this config (is it wrong)?
* Where can I check what happens (no errors in nginx access/error logs!)

Even stripping config down to minimum does not help. Anyone out there with ideas?

Thanks a lot in advance,
Michael

**Comment by rustdesk** [comment]
@21pages follow it up.

**Comment by rustdesk** [comment]
I would suggest you do not do path rewrite like, `location /rustdesk/static/ ` etc. We never tested path rewrite, this is not guaranteed. a subdomain is recommended, `rustdesk.yourcompany.com`.

**Comment by 21pages** [comment]
Doc below works for docker.

https://rustdesk.com/docs/en/self-host/rustdesk-server-pro/faq/#set-up-https-for-web-console-manually

**Reply by MickMaster** [reply]
Thanks for the answer, 21pages !

I saw this instruction - but it did not work as wanted. I created a subdomain (Plesk + DNS) - works. It provides a demo page. If I enter this here:
```
location / {
           proxy_set_header        X-Real-IP       \$remote_addr;
           proxy_set_header        X-Forwarded-For \$proxy_add_x_forwarded_for;
        proxy_pass http://127.0.0.1:21114/;
      }
```
I get this error:

`Invalid nginx configuration: nginx: [emerg] duplicate location "/" in /var/www/vhosts/system/rustdesk.xxx.computer/conf/vhost_nginx.conf:1 nginx: configuration file /etc/nginx/nginx.conf test failed `

I know for sure that there is a html page on /. If I remove the site, Plesk does not offer anything for my nginx config. Any manual config is overwritten later by Plesk. 

Sure - you can call it a Plesk problem now. But it works as any other container on a URL path. Everything works except the web interface. Is there any other way to access it? 

Another strange thing: Other containers make their ports public - so you can configure them in Portainer or Plesk. Rustdesk does not. May be there is a way to modify the config so that the ports can be detected by Plesk?

Thanks a lot for your time,
Mick

**Reply by 21pages** [reply]
Did you try two server blocks for two different domains?
```
server_name rustdesk.example.com;
    location / {
    }
}

server_name plesk.example.com;
    location / {
    }
}
```
![1723126039629](https://github.com/user-attachments/assets/f338cc2f-fede-4e77-87ef-c12e0fadd40e)
